doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        InstalledLocation,
+        /// The standard uninstall command; which may be interactive
+        StandardUninstallCommand,
+        /// An uninstall command that should be non-interactive
+        SilentUninstallCommand,
+        /// The publisher of the package
+        Publisher,
+        /// The locale of the package
+        Locale,
+    };
+
+    /// IMPLEMENTATION NOTE: IPackageVersion from AppInstallerRepositorySearch.h
+    /// A single package version.
+    runtimeclass PackageVersion
+    {
+        PackageVersion();
+
+        /// IMPLEMENTATION NOTE: PackageVersionMetadata fields from AppInstallerRepositorySearch.h
+        /// DESIGN NOTE: These would be a good candidate to leave out in V1. Maybe just keep InstalledLocation and InstalledScope?
+        /// Gets any metadata associated with this package version.
+        /// Primarily stores data on installed packages.
+        /// Metadata fields may have no value (e.g. packages that aren't installed will not have an InstalledLocation).
+        String GetMetadata(PackageVersionMetadata metadataType);
+        /// IMPLEMENTATION NOTE: PackageVersionProperty fields from AppInstallerRepositorySearch.h
+        String Id { get; };
+        String Name { get; };
+        String AppCatalogIdentifier { get; };
+        String AppCatalogName { get; };
+        String Version { get; };
+        String Channel { get; };
+        /// DESIGN NOTE: RelativePath from AppInstallerRepositorySearch.h is excluded as not needed.
+        /// String RelativePath;
+
+        /// IMPLEMENTATION NOTE: PackageVersionMultiProperty fields from AppInstallerRepositorySearch.h
+        /// PackageFamilyName and ProductCode can have multiple values.
+        Windows.Foundation.Collections.IVectorView<String> PackageFamilyName { get; };
+        Windows.Foundation.Collections.IVectorView<String> ProductCode { get; };
+
+        /// Gets the app catalog  where this package version is from.
+        AppCatalog AppCatalog { get; };
+
+        /// DESIGN NOTE:
+        /// GetManifest from IPackageVersion in AppInstallerRepositorySearch is not implemented in V1. That class has a lot of fields and no one requesting it.
+        /// Gets the manifest of this package version.
+        /// virtual Manifest::Manifest GetManifest() = 0;
+    };
+
+    /// IMPLEMENTATION NOTE: PackageVersionKey from AppInstallerRepositorySearch.h
+    /// A key to identify a package version within a package.
+    [version(1)]
+    struct PackageVersionId
+    {
+        /// The app catalog id that this version came from.
+        String AppCatalogId;
+        /// The version.
+        String Version;
+        /// The channel.
+        String Channel;
+    };
+
+    /// IMPLEMENTATION NOTE: IPackage from AppInstallerRepositorySearch.h
+    /// A package, potentially containing information about it's local state and the available versions.
+    runtimeclass Package
+    {
+        Package();

```

Comment:
This should likely not be constructable on its own since it's full of read-only properties.